From 793830be96b95daaa842ab63da66be8840e04315 Mon Sep 17 00:00:00 2001
From: Sanju Rakonde <srakonde@redhat.com>
Date: Mon, 27 Nov 2017 00:18:03 +0530
Subject: [PATCH 135/139] cli: commands are missing in man page

adding missed commands to gluster manual page.

>upstream mainline patch : https://review.gluster.org/#/c/18855

Change-Id: I2e5eb1b3929241275ee7a046c5e3d45a5aa5c4a2
BUG: 1516249
Signed-off-by: Sanju Rakonde <srakonde@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/128885
Reviewed-by: Atin Mukherjee <amukherj@redhat.com>
Tested-by: RHGS Build Bot <nigelb@redhat.com>
---
 doc/gluster.8 | 49 +++++++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 47 insertions(+), 2 deletions(-)

diff --git a/doc/gluster.8 b/doc/gluster.8
index 4c20307..c13a1c2 100644
--- a/doc/gluster.8
+++ b/doc/gluster.8
@@ -35,6 +35,12 @@ The Gluster Console Manager is a command line utility for elastic volume managem
 \fB\ volume info [all|<VOLNAME>] \fR
 Display information about all volumes, or the specified volume.
 .TP
+\fB\ volume list \fR
+List all volumes in cluster
+.TP
+\fB\ volume status [all | <VOLNAME> [nfs|shd|<BRICK>|quotad|tierd]] [detail|clients|mem|inode|fd|callpool|tasks|client-list] \fR
+Display status of all or specified volume(s)/brick
+.TP
 \fB\ volume create <NEW-VOLNAME> [stripe <COUNT>] [replica <COUNT>] [disperse [<COUNT>]] [redundancy <COUNT>] [transport <tcp|rdma|tcp,rdma>] <NEW-BRICK> ... \fR
 Create a new volume of the specified type using the specified bricks and transport type (the default transport type is tcp).
 To create a volume with both transports (tcp and rdma), give 'transport tcp,rdma' as an option.
@@ -51,8 +57,17 @@ Stop the specified volume.
 \fB\ volume set <VOLNAME> <OPTION> <PARAMETER> [<OPTION> <PARAMETER>] ... \fR
 Set the volume options.
 .TP
-\fB\ volume get <VOLNAME> <OPTION/all>\fR
-Get the volume options.
+\fB\ volume get <VOLNAME/all> <OPTION/all> \fR
+Get the value of the all options or given option for volume <VOLNAME> or all option. gluster volume get all all is to get all global options
+.TP
+\fB\ volume reset <VOLNAME> [option] [force] \fR
+Reset all the reconfigured options
+.TP
+\fB\ volume barrier <VOLNAME> {enable|disable} \fR
+Barrier/unbarrier file operations on a volume
+.TP
+\fB\ volume clear-locks <VOLNAME> <path> kind {blocked|granted|all}{inode [range]|entry [basename]|posix [range]} \fR
+Clear locks held on path
 .TP
 \fB\ volume help \fR
 Display help for the volume command.
@@ -70,6 +85,9 @@ If you remove the brick, the data stored in that brick will not be available. Yo
 .B replace-brick
 option.
 .TP
+\fB\ volume reset-brick <VOLNAME> <SOURCE-BRICK> {{start} | {<NEW-BRICK> commit}} \fR
+Brings down or replaces the specified source brick with the new brick.
+.TP
 \fB\ volume replace-brick <VOLNAME> <SOURCE-BRICK> <NEW-BRICK> commit force \fR
 Replace the specified source brick with a new brick.
 .TP
@@ -91,6 +109,15 @@ Locate the log file for corresponding volume/brick.
 .TP
 \fB\ volume log rotate <VOLNAME> [BRICK] \fB
 Rotate the log file for corresponding volume/brick.
+.TP
+\fB\ volume profile <VOLNAME> {start|info [peek|incremental [peek]|cumulative|clear]|stop} [nfs] \fR
+Profile operations on the volume. Once started, volume profile <volname> info provides cumulative statistics of the FOPs performed.
+.TP
+\fB\ volume statedump <VOLNAME> [[nfs|quotad] [all|mem|iobuf|callpool|priv|fd|inode|history]... | [client <hostname:process-id>]] \fR
+Dumps the in memory state of the specified process or the bricks of the volume.
+.TP
+\fB\ volume sync <HOSTNAME> [all|<VOLNAME>] \fR
+Sync the volume information from a peer
 .SS "Peer Commands"
 .TP
 \fB\ peer probe <HOSTNAME> \fR
@@ -102,6 +129,9 @@ Detach the specified peer.
 \fB\ peer status \fR
 Display the status of peers.
 .TP
+\fB\ pool list \fR
+List all the nodes in the pool (including localhost)
+.TP
 \fB\ peer help \fR
 Display help for the peer command.
 .SS "Tier Commands"
@@ -109,9 +139,15 @@ Display help for the peer command.
 \fB\ volume tier <VOLNAME> attach [<replica COUNT>] <NEW-BRICK>... \fR
 Attach to an existing volume a tier of specified type using the specified bricks.
 .TP
+\fB\ volume tier <VOLNAME> start [force] \fR
+Start the tier service for <VOLNAME>
+.TP
 \fB\ volume tier <VOLNAME> status \fR
 Display statistics on data migration between the hot and cold tiers.
 .TP
+\fB\ volume tier <VOLNAME> stop [force] \fR
+Stop the tier service for <VOLNAME>
+.TP
 \fB\ volume tier <VOLNAME> detach start\fR
 Begin detaching the hot tier from the volume. Data will be moved from the hot tier to the cold tier.
 .TP
@@ -164,6 +200,12 @@ Set the hard timeout for the volume. The interval in which limits are retested a
 \fB\ volume quota <VOLNAME> alert-time <TIME> \fR
 Set the frequency in which warning messages need to be logged (in the brick logs) once soft limit is breached.
 .TP
+\fB\ volume inode-quota <VOLNAME> enable/disable \fR
+Enable/disable inode-quota for <VOLNAME>
+.TP
+\fB\ volume quota help \fR
+Display help for volume quota commands
+.TP
 NOTE: valid units of time and their symbols are : hours(h/hr), minutes(m/min), seconds(s/sec), weeks(w/wk), Days(d/days).
 .SS "Geo-replication Commands"
 .TP
@@ -206,6 +248,9 @@ Scrub frequency for volume <VOLNAME>
 \fB\ volume bitrot <VOLNAME> scrub {pause|resume|status|ondemand} \fR
 Pause/Resume scrub. Upon resume, scrubber continues where it left off. status option shows the statistics of scrubber. ondemand option starts the scrubbing immediately if the scrubber is not paused or already running.
 .TP
+\fB\ volume bitrot help \fR
+Display help for volume bitrot commands
+.TP
 .SS "Snapshot Commands"
 .PP
 .TP
-- 
1.8.3.1

